BonusCalcApi/V1/Infrastructure/Migrations/20211112171910_AddSelectableFlagToPayElementTypes.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace V1.Infrastructure.Migrations
+{
+    public partial class AddSelectableFlagToPayElementTypes : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<bool>(
+                name: "selectable",
+                table: "pay_element_types",
+                type: "boolean",
+                nullable: false,
+                defaultValue: false);
+
+            // Set initial values for existing non-editable pay element types
+            migrationBuilder.Sql("UPDATE pay_element_types SET selectable = TRUE WHERE id < 200");
+
+            // Ensure adjustable pay element types are marked as productive
+            migrationBuilder.Sql("UPDATE pay_element_types SET productive = TRUE WHERE adjustment = TRUE");
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
+                name: "selectable",
+                table: "pay_element_types");
+        }
```

Comment:
Just to confirm, you're not undoing the second part of the PR in the down migration, so the update to payelementtypes will be forward-only.

Reply:
Yes, the bonus summary view selects on the productive flag and the adjustment type we envisaged as a non-productive type initially but it's really just an editable productive type - might rename the flag at some point.